### PR TITLE
Replaced `path` package with `upath` to ensure that all paths are normalized

### LINKED ISSUE #254: Tests stopped working on Windows
## 📝 Provide detailed reproduction steps (if any)

Execute tests (`pnpm run test`) on Windows.

### ✔️ Expected result

All tests pass.

### ❌ Actual result

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 5 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  packages/ckeditor5-package-tools/tests/tasks/synchronize-translations.js > lib/tasks/synchronize-translations > synchronizes translation messages (JavaScript)
AssertionError: expected "synchronizeTranslations" to be called with arguments: [ { sourceFiles: [ …(2) ], …(5) } ]

Received:

  1st synchronizeTranslations call:

@@ -1,8 +1,8 @@
  [
    {
-     "corePackagePath": "node_modules/@ckeditor/ckeditor5-core",
+     "corePackagePath": "node_modules\\@ckeditor\\ckeditor5-core",
      "ignoreUnusedCorePackageContexts": true,
      "packagePaths": [
        "/workspace",
      ],
      "skipLicenseHeader": true,


Number of calls: 1

 ❯ packages/ckeditor5-package-tools/tests/tasks/synchronize-translations.js:55:53
     53|
     54|   expect( devTranslations.synchronizeTranslations ).toHaveBeenCalledTimes( 1 );
     55|   expect( devTranslations.synchronizeTranslations ).toHaveBeenCalledWith( {
       |                                                     ^
     56|    // Verify results returned by `globSync()`.
     57|    sourceFiles,

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/5]⎯

 FAIL  packages/ckeditor5-package-tools/tests/tasks/synchronize-translations.js > lib/tasks/synchronize-translations > synchronizes translation messages (TypeScript)
AssertionError: expected "synchronizeTranslations" to be called with arguments: [ { sourceFiles: [ …(2) ], …(5) } ]

Received:

  1st synchronizeTranslations call:

@@ -1,8 +1,8 @@
  [
    {
-     "corePackagePath": "node_modules/@ckeditor/ckeditor5-core",
+     "corePackagePath": "node_modules\\@ckeditor\\ckeditor5-core",
      "ignoreUnusedCorePackageContexts": true,
      "packagePaths": [
        "/workspace",
      ],
      "skipLicenseHeader": true,


Number of calls: 1

 ❯ packages/ckeditor5-package-tools/tests/tasks/synchronize-translations.js:90:53
     88|
     89|   expect( devTranslations.synchronizeTranslations ).toHaveBeenCalledTimes( 1 );
     90|   expect( devTranslations.synchronizeTranslations ).toHaveBeenCalledWith( {
       |                                                     ^
     91|    // Verify results returned by `globSync()`.
     92|    sourceFiles,

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/5]⎯

 FAIL  packages/ckeditor5-package-tools/tests/utils/get-path.js > lib/utils/get-path > getCorePath > returns relative path to the `@ckeditor/ckeditor5-core` package.
AssertionError: expected 'node_modules\@ckeditor\ckeditor5-core' to deeply equal 'node_modules/@ckeditor/ckeditor5-core'

Expected: "node_modules/@ckeditor/ckeditor5-core"
Received: "node_modules\@ckeditor\ckeditor5-core"

 ❯ packages/ckeditor5-package-tools/tests/utils/get-path.js:48:28
     46|     } ) );
     47|
     48|    expect( getCorePath() ).toEqual( 'node_modules/@ckeditor/ckeditor5-core' );
       |                            ^
     49|   } );
     50|  } );

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/5]⎯

 FAIL  packages/ckeditor5-package-tools/tests/utils/get-webpack-config-dll.js > lib/utils/get-webpack-config-dll > calls "getModuleResolutionPaths" with correct arguments
AssertionError: expected false to deeply equal true

- Expected
+ Received

- true
+ false

 ❯ packages/ckeditor5-package-tools/tests/utils/get-webpack-config-dll.js:152:91
    150|   const [ firstArgument ] = getModuleResolutionPaths.mock.calls[ 0 ];
    151|
    152|   expect( firstArgument.endsWith( '/packages/ckeditor5-package-tools/lib/utils/../..' ) ).toEqual( true );
       |                                                                                           ^
    153|  } );
    154|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/5]⎯

 FAIL  packages/ckeditor5-package-tools/tests/utils/get-webpack-config-server.js > lib/utils/get-webpack-config-server > calls "getModuleResolutionPaths" with correct arguments
AssertionError: expected false to deeply equal true

- Expected
+ Received

- true
+ false

 ❯ packages/ckeditor5-package-tools/tests/utils/get-webpack-config-server.js:123:91
    121|   const [ firstArgument ] = getModuleResolutionPaths.mock.calls[ 0 ];
    122|
    123|   expect( firstArgument.endsWith( '/packages/ckeditor5-package-tools/lib/utils/../..' ) ).toEqual( true );
       |                                                                                           ^
    124|  } );
    125|
```

### ❓ Possible solution

It stopped working somewhere between [v4.0.1](https://github.com/ckeditor/ckeditor5-package-generator/releases/tag/v4.0.1) and [v4.0.2](https://github.com/ckeditor/ckeditor5-package-generator/releases/tag/v4.0.2).

## 📃 Other details

* Node: …
* Npm / Yarn / Pnpm: …
* First affected version: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by psmyrek**
### 🚀 Summary

Replaced `path` package with `upath` to ensure that all paths are normalized.

---

### 📌 Related issues

* Closes #254

---

### 💡 Additional information

*Optional: Notes on decisions, edge cases, or anything helpful for reviewers.*

### CODE REVIEW

**Review by filipsobol — DISMISSED** [review]
LGTM, although I don't have a Windows machine to test it.

I posted a single nitpick above.

**Review by filipsobol — APPROVED** [review]
